okhttp/src/main/java/okhttp3/internal/connection/RouteException.kt

```diff
@@ -13,36 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package okhttp3.internal.connection;
+package okhttp3.internal.connection
 
-import java.io.IOException;
-
-import static okhttp3.internal.Util.addSuppressedIfPossible;
+import okhttp3.internal.Util.addSuppressedIfPossible
+import java.io.IOException
 
 /**
  * An exception thrown to indicate a problem connecting via a single Route. Multiple attempts may
  * have been made with alternative protocols, none of which were successful.
  */
-public final class RouteException extends RuntimeException {
-  private IOException firstException;
-  private IOException lastException;
-
-  RouteException(IOException cause) {
-    super(cause);
-    firstException = cause;
-    lastException = cause;
-  }
-
-  public IOException getFirstConnectException() {
-    return firstException;
-  }
+class RouteException internal constructor(val firstConnectException: IOException) :
+    RuntimeException(firstConnectException) {
+  var lastConnectException: IOException? = null
+    private set
 
-  public IOException getLastConnectException() {
-    return lastException;
+  init {
+    lastConnectException = firstConnectException
   }
 
-  void addConnectException(IOException e) {
-    addSuppressedIfPossible(firstException, e);
-    lastException = e;
+  fun addConnectException(e: IOException) {
+    addSuppressedIfPossible(firstConnectException, e)
```

Comment:
Sorry I conflicted you here by removing this method. We can use `addSuppressed` directly now.